nztides_app/app/src/main/java/com/palliser/nztides/NZTides.java

```diff
@@ -109,7 +109,12 @@ public String calc_outstring(String port){
 			SimpleDateFormat dformat = new SimpleDateFormat(
 					"HH:mm E dd/MM/yy zzz");
 
-	    	DataInputStream tidedat = new DataInputStream(am.open(port+".tdat",1));
+		String[] files = am.list("/");
+		String[] lowerCaseFiles = Arrays.stream(files).map(String::toLowerCase).toArray(String[]::new);
+		int indexOfPort = Arrays.asList(lowerCaseFiles).indexOf(port.toLowerCase() + ".tdat");
+		if (indexOfPort == -1) { throw new IOException("port not found even when matched case in use"); }
+		String exactPortFilename = files[indexOfPort];
+	    	DataInputStream tidedat = new DataInputStream(am.open(exactPortFilename,1));
```

Comment:
Well i would extract a method for the loadTideData to encapsulated this lookup and stream creation

You could filter on stream and then use an optional findFirst and then test that, rather than loading the whole list and then searching it.

 Something like this...

   String[] files = am.list("/");

        String exactPortFilename = Arrays.stream(files)
                .filter(f -> f.equalsIgnoreCase(port + ".tdat"))
                .findFirst()
                .orElseThrow(() -> new IOException("Port not found even when matched case in use"));

        DataInputStream tidedat = new DataInputStream(am.open(exactPortFilename, 1));

Reply:
maybe a little test to validate the extracted method works as ezxpected ;)